virtual_rainforest/models/abiotic/atmospheric_co2.py

```diff
@@ -0,0 +1,235 @@
+        initial_co2_profile=initial_co2_profile.isel(atmosphere_layers=-1),
+        soil_respiration=soil_respiration,
+        animal_respiration=animal_respiration,
+    )
+    # Mix CO2
+    co2_profile_mixed = vertical_mixing_co2(
+        co2_above_canopy=initial_co2_profile.isel(atmosphere_layers=0),
+        co2_within_canopy=co2_within_canopy,
+        co2_below_canopy=co2_below_canopy,
+        mixing=False,
+    )
+
+    return co2_profile_mixed
+
+
+# helper functions
+def initialise_co2_profile(
+    ambient_atmospheric_co2: DataArray,
+    atmosphere_layers: int,
+    initialisation_method: str = "homogenous",
+) -> DataArray:
+    r"""Initialise CO\ :sub:`2`\ profile.
+
+    Args:
+        ambient_atmospheric_co2: ambient CO\ :sub:`2`\ concentraion, [ppm],
+        atmosphere_layers: number of atmosphere layers for which CO\ :sub:`2`\
+             concentration is calculated
+        initialisation_method: interpolation method for initial CO\ :sub:`2`\ profile,
+            default copies ambient CO\ :sub:`2`\ to all vertical levels
+
+    Returns:
+        initial vertical CO\ :sub:`2`\ profile, [ppm]
+    """
+
+    if initialisation_method != "homogenous":
+        raise (NotImplementedError("This method is not implemented"))
+
+    else:
+        initial_co2_profile = DataArray(
+            ambient_atmospheric_co2.expand_dims(
+                dim={"atmosphere_layers": atmosphere_layers}
+            )
+        )
+    return initial_co2_profile
+
+
+def calculate_co2_within_canopy(
+    initial_co2_profile: DataArray,
+    plant_net_co2_assimilation: DataArray,
+) -> DataArray:
+    r"""Calculate CO\ :sub:`2`\ concentration within canopy.
+
+    This function subtracts the net CO\ :sub:`2`\ assimilation of plants from the
+    ambient CO\ :sub:`2`\ level. Make sure that the initial_co2_profile has the same
+    dimensions as the canopy.
+
+    Args:
+        initial_co2_profile: initial CO\ :sub:`2`\ profile, [ppm]
+        plant_net_co2_assimilation: plant net canron assimilation, [ppm]
+
+    Returns:
+        CO\ :sub:`2`\ concentration within canopy, [ppm]
+    """
+
+    return initial_co2_profile - plant_net_co2_assimilation
+
+
+def calculate_co2_below_canopy(
+    initial_co2_profile: DataArray,
+    soil_respiration: DataArray,
+    animal_respiration: DataArray,
+) -> DataArray:
+    r"""Calculate CO\ :sub:`2`\ concentration below canopy.
+
+    This function adds the net respiration of soil organisms and animals to the ambient
+    CO\ :sub:`2`\ level. Make sure that the initial_co2_profile has the same dimensions
+    as layers below canopy.
+
+    Args:
+        initial_co2_profile: initial CO\ :sub:`2`\ profile, [ppm]
+        soil_respiration: soil respiration, [ppm]
+        animal_respiration: animal respiration, [ppm]
+
+    Returns:
+        CO\ :sub:`2`\ concentration below canopy, [ppm]
+    """
+    return initial_co2_profile + soil_respiration + animal_respiration
+
+
+def vertical_mixing_co2(
```

Comment:
I probably would add a test here because it's probably worth testing that the Exception is raised under the right conditions, also I think it is worth explicitly testing that the order in the `co2_profile_mixed` is as expected